include/ensmallen_bits/age_moea/agemoea_impl.hpp

```diff
@@ -0,0 +1,781 @@
+    FindExtremePoints<MatType>(extreme, calculatedObjectives, front);
+    NormalizeFront<MatType>(calculatedObjectives, normalize, front, extreme);
+
+    for (size_t index = 0; index < front.size(); index++)
+    {
+      calculatedObjectives[front[index]] = calculatedObjectives[front[index]] / normalize;
+    }
+
+    dimension = GetGeometry<MatType>(calculatedObjectives, front,
+                                           extreme);
+
+    std::set<size_t> selected;
+    std::set<size_t> remaining;
+    
+    // Create the selected and remaining sets.
+    for (size_t index: extreme)
+    { 
+      selected.insert(index);
+      survivalScore[front[index]] = arma::datum::inf;
+    }
+    for (size_t i = 0; i < front.size(); i++)
+    {
+      if (selected.count(i) == 0)
+      {
+        remaining.insert(i);
+      }
+    }
+
+    arma::Mat<ElemType> pairwise(front.size(), front.size(), arma::fill::zeros);
+    PairwiseDistance<MatType>(pairwise,calculatedObjectives,front,dimension);
+    arma::Row<typename MatType::elem_type> proximity(front.size(), arma::fill::zeros);
+    arma::Row<typename MatType::elem_type> diversity(front.size(), arma::fill::zeros);
+    arma::Row<typename MatType::elem_type> value(front.size(), arma::fill::zeros);
+    
+    // Calculate the diversity and proximity score.
+    for (size_t i = 0; i < front.size(); i++)
+    {
+      proximity[i] = std::pow(arma::accu(arma::pow(
+        arma::abs(calculatedObjectives[front[i]]), dimension)), 1.0 / dimension);
+    }
+    
+    while (remaining.size() > 0)
+    {
+      std::set<size_t>::iterator it;
+      for (it = remaining.begin(); it != remaining.end(); it++)
+      {
+        diversity[*it] = DiversityScore<MatType>(selected, pairwise, *it);
+        value[*it] = diversity[*it] / proximity[*it];
+      }
+      size_t index = arma::index_max(value);
+      survivalScore[front[index]] = value[index];
+      value[index] = - 1;
+      selected.insert(index);
+      remaining.erase(index);
+    }
+  }
+
+  // Calculations for the other fronts.
+  else
+  {
+    for(size_t i = 0; i < front.size(); i++)
+    {
+      calculatedObjectives[front[i]] = calculatedObjectives[front[i]] / normalize;
+      survivalScore[front[i]] =  std::pow(arma::accu(arma::pow(arma::abs(
+        calculatedObjectives[front[i]] - idealPoint), dimension)), 1.0 / dimension);
+    }
+
+  }
+}
+
+//! Comparator for survival score based sorting.
+template<typename MatType>
+inline bool AGEMOEA::SurvivalScoreOperator(size_t idxP,
+                                    size_t idxQ,
+                                    const std::vector<size_t>& ranks,
+                                    const std::vector<typename MatType::elem_type>& survivalScore)
+{
+  if (ranks[idxP] < ranks[idxQ])
+    return true;
+  else if (ranks[idxP] == ranks[idxQ] && survivalScore[idxP] > survivalScore[idxQ])
+    return true;
+
+  return false;
+}
+
+} // namespace ens
+
+#endif
```

Comment:
This is a minor issue too, but if you can make sure files have newlines at the end of them, I would appreciate it 👍